Open Judge System/LocalWorker/OJS.LocalWorker/OjsSubmissionProcessingStrategy.cs

```diff
@@ -41,14 +41,15 @@ public class OjsSubmissionProcessingStrategy : SubmissionProcessingStrategy<int>
             this.submissionsForProcessingData = submissionsForProcessingData;
         }
 
-        public override IOjsSubmission RetrieveSubmission()
+        public override IOjsSubmission RetrieveSubmission(WorkerType workerType)
         {
             lock (this.SubmissionsForProcessing)
             {
                 if (this.SubmissionsForProcessing.IsEmpty)
                 {
                     this.submissionsForProcessingData
                         .GetAllUnprocessed()
+                        .Where(s => s.WorkerType == workerType)
```

Comment:
Is there a chance to have submissions that will not be picked up with this filtration in any worker? In other words stay pending forever?

Reply:
No, when the migration is applied all 3 (Contest, Submission and SubmissionForProcessing will have column "WorkerType" with default value 0 (Legacy) ).

Reply:
Is this still a valid response after the updates?

Reply:
No, currently if the submission's Worker type is Default, then it will check the related SubmissionType's Worker Type. If it is Default, then it will look in the contest. If the contest's Worker Type is Default, then it will set it to the one that is configured in the App Settings. Currently it is Legacy